spec.bs

```diff
@@ -491,12 +494,236 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 
 A {{URLPattern}} is associated with multiple <dfn>component</dfn> [=structs=].
 
-A [=component=] has an associated <dfn for=component>pattern string</dfn>, a string, initially null.
+A [=component=] has an associated <dfn for=component>pattern string</dfn>, a [=pattern string/well formed=] [=/pattern string=], initially null.
 
 A [=component=] has an associated <dfn for=component>regular expression</dfn>, a {{RegExp}}, initially null.
 
 A [=component=] has an associated <dfn for=component>group name list</dfn>, an array of strings, initially an empty array.
 
+<h2 id=patterns>Patterns</h2>
+
+A <dfn>pattern string</dfn> is a string that is written to match a set of target strings.  A <dfn for="pattern string">well formed</dfn> pattern string conforms to a particular pattern syntax.  This pattern syntax is directly based on the syntax used by the popular [path-to-regexp](https://github.com/pillarjs/path-to-regexp) JavaScript library.
+
+<h3 id=parsing-patterns>Parsing Patterns</h3>
+
+<h4 id=tokens>Tokens</h4>
+
+A <dfn>token list</dfn> is a [=list=] containing zero or more [=token=] [=structs=].
+
+A <dfn>token</dfn> is a [=struct=] representing a single lexical token within a [=/pattern string=].
+
+A [=token=] has an associated <dfn for=token>type</dfn>, a string, initially "<a for=token/type>`invalid-char`</a>".  It must be one of the following:
+
+<dl>
+  <dt>"<dfn for=token/type>`open`</dfn>"</dt>
+  <dd>The [=token=] represents a U+007B (`{`) code point.
+  <dt>"<dfn for=token/type>`close`</dfn>"</dt>
+  <dd>The [=token=] represents a U+007D (`}`) code point.
+  <dt>"<dfn for=token/type>`regex`</dfn>"</dt>
```

Comment:
Oh, I forgot this one last time: it's traditional to abbreviate regular expressions as "regexp", not "regex", in the web ecosystem.